lib/galaxy_test/api/test_workflows.py

```diff
@@ -353,6 +353,76 @@ def test_index_hidden(self):
         workflow_index = self._get("workflows?show_hidden=false").json()
         assert not [w for w in workflow_index if w["id"] == workflow_id]
 
+    def test_index_ordering(self):
+        # ordered by update_time on the stored workflows with all user's workflows
+        # before workflows shared with user.
+        my_workflow_id_1 = self.workflow_populator.simple_workflow("mine_1")
+        my_workflow_id_2 = self.workflow_populator.simple_workflow("mine_2")
+        my_email = self.dataset_populator.user_email()
+        with self._different_user():
+            their_workflow_id_1 = self.workflow_populator.simple_workflow("theirs_1")
+            their_workflow_id_2 = self.workflow_populator.simple_workflow("theirs_2")
+            self.workflow_populator.share_with_user(their_workflow_id_1, my_email)
+            self.workflow_populator.share_with_user(their_workflow_id_2, my_email)
+        index_ids = self.workflow_populator.index_ids()
+        worklfow_index_response = self._get("workflows")
+        worklfow_index_response.raise_for_status()
+        index_ids = [w["id"] for w in worklfow_index_response.json()]
```

Comment:
These 3 lines seem redundant to the previous one

```suggestion
```

Reply:
I've rebased with this fix, thanks for the catch!